eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Tools.hs

```diff
@@ -162,7 +156,7 @@ evaluateTransactionExecutionUnits pp tx utxo ei sysS costModels = do
     Nothing ->
       let getInfo :: Language -> Either TranslationError (Language, VersionedTxInfo)
           getInfo lang = (,) lang <$> txInfo pp lang ei sysS utxo tx
-          txInfos = map getInfo (Set.toList $ languagesUsed (Map.elems scripts))
+          txInfos = [getInfo lang | (_, lang, _) <- neededAndConfirmedToBePlutus]
```

Comment:
This will force `getInfo` to be called as many times as there are scripts, instead of as many times as there are languages. In order to fix this we need to nub it:
```suggestion
          txInfos = map getInfo $ Set.toList $
                      Set.fromList [lang | (_, lang, _) <- neededAndConfirmedToBePlutus]
```